Tests and encryption

m-iwanicki opened the pull request:
Before (testing debug image so root account is unlocked and this test fails):

```
genericx86-64:~$ sudo cukinia
```

**Results:**

```
[PASS]  Check if cukinia is run with root privileges
----> encryption.conf <----
----> Encryption feature <----
Test rootfs partition A
[PASS]  Check if /dev/disk/by-label/rootfs_a is encrypted
[PASS]  Check if we can get rootfs LUKS device information dump
[FAIL]  Check if rootfs can only be decrypted with TPM2 or recovery key
[FAIL]  Check if rootfs can be decrypted via TPM2
[FAIL]  Check if rootfs can be decrypted via recovery key
Test rootfs partition B
[PASS]  Check if /dev/disk/by-label/rootfs_b is encrypted
[PASS]  Check if we can get rootfs LUKS device information dump
[FAIL]  Check if rootfs can only be decrypted with TPM2 or recovery key
[FAIL]  Check if rootfs can be decrypted via TPM2
[FAIL]  Check if rootfs can be decrypted via recovery key
Test rootfs overlay partition
[PASS]  Check if /dev/disk/by-label/rwoverlay is encrypted
[PASS]  Check if we can get rootfs LUKS device information dump
[FAIL]  Check if rootfs can only be decrypted with TPM2 or recovery key
[FAIL]  Check if rootfs can be decrypted via TPM2
[FAIL]  Check if rootfs can be decrypted via recovery key
----> hardening.conf <----
----> Users <----
[PASS]  Checking user "user" exists
[PASS]  Checking user "user" is member of "user sudo"
[FAIL]  Check if root account is locked
----> Kernel lockdown feature <----
[PASS]  Check if access to /dev/mem is restricted
[PASS]  Check if kernel lockdown is enabled
[PASS]  Check dmesg for blocked access to /dev/mem
----> otab.conf <----
----> A/B feature <----
[PASS]  Check if boot partition A exists
[PASS]  Check if boot partition B exists
[PASS]  Check if rootfs partition A exists
[PASS]  Check if rootfs partition B exists
[PASS]  Check if rwoverlay partition exists
[PASS]  Check if / is mounted as overlay
[PASS]  Check if /media/rfs/ro is read-only
[PASS]  Check if correct partition (A) is mounted on /media/rfs/ro
[PASS]  Check if swupdate is configured to use efivar bootloader
[PASS]  Checking if systemd unit "initramfs-shutdown.service" is active
result: 10 failure(s)
```

After:

<img width="691" height="660" alt="image" src="https://github.com/user-attachments/assets/3921681b-c0f0-4b99-aa4d-2f2c5ee21bb3" />

<details><summary>Text log</summary>
<p>

```
[PASS]  Check if cukinia is run with root privileges
----> encryption.conf <----
----> Encryption feature <----
----> Test rootfs partition A <----
[PASS]  Check if /dev/disk/by-label/rootfs_a is encrypted
[PASS]  Check if we can get 'rootfs partition A' LUKS device information dump
----> Test rootfs partition B <----
[PASS]  Check if /dev/disk/by-label/rootfs_b is encrypted
[PASS]  Check if we can get 'rootfs partition B' LUKS device information dump
----> Test rootfs overlay partition <----
[PASS]  Check if /dev/disk/by-label/rwoverlay is encrypted
[PASS]  Check if we can get 'rootfs overlay partition' LUKS device information dump
----> hardening.conf <----
----> Users <----
[PASS]  Checking user "user" exists
[PASS]  Checking user "user" is member of "user sudo"
[FAIL]  Check if root account is locked
----> Kernel lockdown feature <----
[PASS]  Check if access to /dev/mem is restricted
[PASS]  Check if kernel lockdown is enabled
[PASS]  Check dmesg for blocked access to /dev/mem
----> otab.conf <----
----> A/B feature <----
[PASS]  Check if boot partition A exists
[PASS]  Check if boot partition B exists
[PASS]  Check if rootfs partition A exists
[PASS]  Check if rootfs partition B exists
[PASS]  Check if rwoverlay partition exists
[PASS]  Check if / is mounted as overlay
[PASS]  Check if /media/rfs/ro is read-only
[PASS]  Check if correct partition (A) is mounted on /media/rfs/ro
[PASS]  Check if swupdate is configured to use efivar bootloader
[PASS]  Checking if systemd unit "initramfs-shutdown.service" is active
result: 1 failure(s)
```

</p>
</details> 

Removed tests:

```
[FAIL]  Check if rootfs can only be decrypted with TPM2 or recovery key
[FAIL]  Check if rootfs can be decrypted via TPM2
[FAIL]  Check if rootfs can be decrypted via recovery key 
```

because:

- recovery key - we use password instead of recovery key now
- TPM2 - enrolling is done manually after OS/firmware preparation and is optional. If TPM tests are needed then they should be done in e.g. OSFV.

---

Also fixes TPM2 errors if PCRs change e.g. when rerunning QEMU:

```
Starting systemd-udevd version 255.13^
Enter disk encryption password:
Mounting rw filesystem using overlay...

Distro for Zarhus product 0.1.0 genericx86-64 ttyS0

genericx86-64 login: user
Password:

genericx86-64:~$ sudo cryptsetup luksDump /dev/sda3 | grep -i tpm2 -B1
Tokens:
  0: systemd-tpm2
```

To make sure that those errors were really there, just hidden I tried to decrypt partition manually, without redirecting stderr output to `/dev/null` and I got expected errors:

```
genericx86-64:~$ sudo cryptsetup open /dev/sda4 luks_rootfs_b
WARNING:esys:/usr/src/debug/tpm2-tss/4.0.2/src/tss2-esys/api/Esys_Load.c:324:Esys_Load_Finish() Received TPM Error
ERROR:esys:/usr/src/debug/tpm2-tss/4.0.2/src/tss2-esys/api/Esys_Load.c:112:Esys_Load() Esys Finish ErrorCode (0x0000018b)
Failed to unseal secret using TPM2: State not recoverable
Enter passphrase for /dev/sda4:
```

---

Results for meta-zarhus image without A/B update feature:

```
[PASS]  Check if cukinia is run with root privileges
----> encryption.conf <----
----> Encryption feature <----
----> Test rootfs <----
[PASS]  Check if /dev/disk/by-label/rootfs is encrypted
[PASS]  Check if we can get 'rootfs' LUKS device information dump
----> hardening.conf <----
----> Users <----
[PASS]  Checking user "user" exists
[PASS]  Checking user "user" is member of "user sudo"
[FAIL]  Check if root account is locked
----> Kernel lockdown feature <----
[PASS]  Check if access to /dev/mem is restricted
[PASS]  Check if kernel lockdown is enabled
[PASS]  Check dmesg for blocked access to /dev/mem
result: 1 failure(s)
```

3mkusiak reviewed (commented):
LGTM

Tested this, logs in Jira task.

Please fix `pre-commit` issues, so we can merge this.

m-iwanicki commented:
Fixed, updated hooks and fixed one remaining issue.
Pre-commit still fails due to rust. We need to finally fix this (current state: https://github.com/zarhus/meta-zarhus/pull/66)
